cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go

```diff
@@ -108,7 +108,7 @@ func (aws *awsCloudProvider) NodeGroupForNode(node *apiv1.Node) (cloudprovider.N
 	asg := aws.awsManager.GetAsgForInstance(*ref)
 
 	if asg == nil {
-		return nil, nil
+		return nil, fmt.Errorf("cannot find ASG for node %v", ref.Name)
```

Comment:
We return an error here, but this method gets called looking for Kubernetes nodes that aren't part of a discovered ASG (in my case, my master machines are part of an ASG, but not an ASG that the cluster autoscaler discovers or touches).

In `core.getNodeInfosForGroups`, we call on this method, and any errors that come back out now cause that function to fail earlier than it used to. It was written to accommodate the nil return here, without an accompanying error.

See #2345 for what may be a consequence of this change (with initial commentary from @losipiuk).